src/webgpu/api/validation/createView.spec.ts

```diff
@@ -1,311 +1,318 @@
-    { arrayLayerCount: ARRAY_LAYER_COUNT_2D, _success: true }, // it is OK to create a 2D array texture view on a 2D array texture
-    // baseArrayLayer == k && arrayLayerCount == 0 means to use layers k..end.
-    { arrayLayerCount: 0, baseArrayLayer: 0, _success: true },
-    { arrayLayerCount: 0, baseArrayLayer: 1, _success: true },
-    { arrayLayerCount: 0, baseArrayLayer: ARRAY_LAYER_COUNT_2D - 1, _success: true },
-    { arrayLayerCount: 0, baseArrayLayer: ARRAY_LAYER_COUNT_2D, _success: false },
-    // It is an error for the array layer range of the view to exceed that of the texture
-    { arrayLayerCount: ARRAY_LAYER_COUNT_2D + 1, baseArrayLayer: 0, _success: false },
-    { arrayLayerCount: ARRAY_LAYER_COUNT_2D, baseArrayLayer: 1, _success: false },
-    { arrayLayerCount: 2, baseArrayLayer: ARRAY_LAYER_COUNT_2D - 1, _success: false },
-    { arrayLayerCount: 1, baseArrayLayer: ARRAY_LAYER_COUNT_2D, _success: false },
-  ])
   .fn(async t => {
-    const { dimension = '2d-array', arrayLayerCount, baseArrayLayer, _success } = t.params;
+    const { format, aspect } = t.params;
+    await t.selectDeviceForTextureFormatOrSkipTestCase(format);
+    const info = kTextureFormatInfo[format];
 
-    const texture = t.createTexture({ arrayLayerCount: ARRAY_LAYER_COUNT_2D });
-
-    const descriptor = t.getDescriptor({
-      dimension,
-      arrayLayerCount,
-      baseArrayLayer,
+    const texture = t.device.createTexture({
+      format,
+      size: [4, 4, 1],
+      usage: GPUTextureUsage.SAMPLED,
     });
 
+    const success =
+      aspect === 'all' ||
+      (aspect === 'depth-only' && info.depth) ||
+      (aspect === 'stencil-only' && info.stencil);
     t.expectValidationError(() => {
-      texture.createView(descriptor);
-    }, !_success);
+      texture.createView({ aspect });
+    }, !success);
   });
 
-g.test('Using_defaults_validates_the_same_as_setting_values_for_more_than_1_array_layer')
-  .paramsSimple([
-    { _success: true },
-    { format: 'rgba8unorm', _success: true },
-    { format: 'r8unorm', _success: false },
-    { dimension: '2d-array', _success: true },
-    { dimension: '2d', _success: true }, // setting dimension to 2D means that arrayLayerCount will default to 1.
-    { arrayLayerCount: ARRAY_LAYER_COUNT_2D, _success: false }, // setting array layers to non-0 means the dimensionality will default to 2D so by itself it causes an error.
-    { arrayLayerCount: ARRAY_LAYER_COUNT_2D, dimension: '2d-array', _success: true },
-    {
-      arrayLayerCount: ARRAY_LAYER_COUNT_2D,
-      dimension: '2d-array',
-      mipLevelCount: MIP_LEVEL_COUNT,
-      _success: true,
-    },
-  ] as const)
-  .fn(async t => {
-    const { format, dimension, arrayLayerCount, mipLevelCount, _success } = t.params;
+g.test('2d_dimension_layers')
+  .desc(
+    `For the 2d view dimensions {2d, 2d-array, cube, cube-array}, test validation of layer counts:
+  - 2d must have 1 layer
```

Comment:
This can be: `1d, 2d, and 3d must have 1 layer`.
Also rename this function to dimension_layers and add test for `1d` and `3d` (at least add test for 3d, which can be supported in WebGPU MVP)?

Reply:
It's not possible to specify layers for 3d, but good idea to add tests for 1d. It may be a separate test since it'll be very small.

Reply:
Actually there is something more general to test here, which is that any dimension being 0 on any texture view is invalid.

Reply:
oops ignore all that, I forgot how texture views work, brain got stale over the weekend